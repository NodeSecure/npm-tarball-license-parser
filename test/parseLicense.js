"use strict";

// Require Node.js Dependencies
const { join } = require("path");

// Require Internal Dependencies
const parseLicense = require("..");

// CONSTANTS
const FIXTURE_PATH = join(__dirname, "fixtures");

test("parseLicense of project1", async() => {
    const result = await parseLicense(join(FIXTURE_PATH, "project1"));
    expect(result.uniqueLicenseIds).toEqual(new Set(["ISC", "MIT"]));
    expect(result).toMatchSnapshot();
});
