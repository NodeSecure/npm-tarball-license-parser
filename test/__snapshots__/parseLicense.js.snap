// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseLicense of project1 1`] = `
Object {
  "hasMultipleLicenses": true,
  "licenses": Array [
    Object {
      "from": "package.json",
      "spdx": Object {
        "fsf": true,
        "fsfAndOsi": true,
        "includesDeprecated": false,
        "osi": true,
      },
      "spdxLicenseLinks": Array [
        "https://spdx.org/licenses/ISC.html#licenseText",
      ],
      "uniqueLicenseIds": Array [
        "ISC",
      ],
    },
    Object {
      "from": "LICENSE",
      "spdx": Object {
        "fsf": true,
        "fsfAndOsi": true,
        "includesDeprecated": false,
        "osi": true,
      },
      "spdxLicenseLinks": Array [
        "https://spdx.org/licenses/MIT.html#licenseText",
      ],
      "uniqueLicenseIds": Array [
        "MIT",
      ],
    },
  ],
  "uniqueLicenseIds": Array [
    "ISC",
    "MIT",
  ],
}
`;
